### PR TITLE
Issue #75: Major refactor of model - `Person` to `Candidate`

### LINKED ISSUE #75: Refactor all occurrences of `Person` to `Candidate`

### PR DESCRIPTION
**Pull request by domlimm**
This PR consists of refactoring all the files related to the previous `Person` model to now `Candidate` model.

@charmainehly @LeeEnEn @lzan98 @tiewweijian hey fam, don't mind once this PR has been approved and merged, please do a quick test on your assigned command i.e. Our 5 basic commands, to ensure everything works as expected.

I ran just the basic test of each of our command, and it works as expected. All test cases passed too.

It's a really big PR, I scanned through each file one by one to make sure it's logically working based on the codes itself, but need your help to test.

I guess any 2 of you can just review and approve. Or if y'all want, 4 of you can too.

Sorry for the troubles, and thanks y'all! 🥂💟 

Closes #75.

### CODE REVIEW

**Review by lzan98 — APPROVED** [review]
LGTM! I think we should merge this soon so that future PRs have less conflicts to resolve! 

Did a quick scan and looks good (since it's mostly refactoring)

**Comment by charmainehly** [comment]
Thanks @domlimm for refactoring!! :)) Hmmm, actually I'm noticing a number of merge conflicts due to the refactoring. Perhaps we should merge in the previous PRs first if they are good to go, so that we can avoid multiple corrections of similar merge conflicts?

File: src/main/java/seedu/address/logic/commands/AddCommand.java
Comment (LeeEnEn) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:37]:
```suggestion
    public static final String MESSAGE_SUCCESS = "New candidate added: %1$s";
    public static final String MESSAGE_DUPLICATE_PERSON = "This candidate already exists in the address book";
```

Comment (LeeEnEn) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:37]: Perhaps the `person` on these two lines should be changed to `candidate`.

**Review by LeeEnEn — COMMENTED** [review]
Minor changes needed for these two lines. Otherwise everything else looks good!
Good job on the tough refactoring!

File: src/main/java/seedu/address/logic/commands/AddCommand.java
Comment (lzan98) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:37]: nice catch! The String could also be changed to MESSAGE_DUPLICATE_CANDIDATE.

Comment (LeeEnEn) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:37]: You are right Leon! How did I missed that 🤣

**Comment by domlimm** [comment]
> LGTM! I think we should merge this soon so that future PRs have less conflicts to resolve!
> 
> Did a quick scan and looks good apart from the merge conflicts

Thanks for reviewing @lzan98! Yes the merge conflicts came after #81.

**Comment by domlimm** [comment]
> Thanks @domlimm for refactoring!! :)) Hmmm, actually I'm noticing a number of merge conflicts due to the refactoring. Perhaps we should merge in the previous PRs first if they are good to go, so that we can avoid multiple corrections of similar merge conflicts?

Hey @charmainehly, yup I believe we should merge in the rest first then I can fix the conflicts further. Maybe we set a cut off time tomorrow for me to have ample time to fix the conflicts before our demo.

@LeeEnEn @lzan98 @tiewweijian let us all discuss a cut-off time on thu in tele.

> * @domlimm I'm not sure if the refactoring takes into account the new sort command, if it doesn't then perhaps some changes will need to be made to PR [Add sort command to sort across all candidate fields (Issue #72) #76](https://github.com/AY2122S2-CS2103-F11-2/tp/pull/76) as well depending on the order of merge! :)

@charmainehly yes, since #76 isn't merged in yet, there's no refactoring done on the files in the PR. Will make the changes as soon as it's merged in. Probably wait for the rest of the PRs to merge in too before fixing etc.

**Review by LeeEnEn — APPROVED** [review]

**Review by tiewweijian — APPROVED** [review]
All looks good to me ! Agree with the rest I think we can set a deadline before merging this PR in

**Pull request merged**
